lib/archethic/utils/hydrating_cache/cache_entry.ex

```diff
@@ -0,0 +1,229 @@
+defmodule Archethic.Utils.HydratingCache.CacheEntry.StateData do
+  @moduledoc """
+  Struct describing the state of a cache entry FSM
+  """
+
+  defstruct running_func_task: :undefined,
+            hydrating_func: :undefined,
+            getters: [],
+            ttl: :undefined,
+            refresh_interval: :undefined,
+            key: :undefined,
+            value: :"$$undefined",
+            timer_func: :undefined,
+            timer_discard: :undefined
+end
+
+defmodule Archethic.Utils.HydratingCache.CacheEntry do
+  @moduledoc """
+  This module is a finite state machine implementing a cache entry.
+  There is one such Cache Entry FSM running per registered key.
+
+  It is responsible for :
+  - receiving request to get the value for the key it is associated with
+  - Run the hydrating function associated with this key
+  - Manage various timers
+  """
+  alias Archethic.Utils.HydratingCache.CacheEntry
+  @behaviour :gen_statem
+
+  use Task
+  require Logger
+
+  def start_link([fun, key, ttl, refresh_interval]) do
+    :gen_statem.start_link(__MODULE__, [fun, key, ttl, refresh_interval], [])
+  end
+
+  @impl :gen_statem
+  def init([fun, key, ttl, refresh_interval]) do
+    timer = :timer.send_interval(refresh_interval, self(), :hydrate)
+
+    ## Hydrate the value
+    {:ok, :running,
+     %CacheEntry.StateData{
+       timer_func: timer,
+       hydrating_func: fun,
+       key: key,
+       ttl: ttl,
+       refresh_interval: refresh_interval
+     }}
+  end
+
+  @impl :gen_statem
+  def callback_mode, do: [:handle_event_function, :state_enter]
+
+  @impl :gen_statem
+  def handle_event({:call, from}, {:get, _from}, :idle, data) do
+    IO.puts("==> Get in idle state")
```

Comment:
maybe we can use logger to be able to adapt the level of logs in dev for example

Reply:
yes, there are still some traces I need to remove